src/services/listCourtCases.ts

```diff
@@ -134,26 +143,55 @@ const listCourtCases = async (
             qb.orWhere(
               new Brackets((dateRangeQuery) => {
                 dateRangeQuery
-                  .andWhere({ courtDate: MoreThanOrEqual(dateRange.from) })
-                  .andWhere({ courtDate: LessThanOrEqual(dateRange.to) })
+                  .andWhere({
+                    courtDate: MoreThanOrEqual(dateRange.from)
+                  })
+                  .andWhere({
+                    courtDate: LessThanOrEqual(dateRange.to)
+                  })
               })
             )
           })
         })
       )
     } else {
       query
-        .andWhere({ courtDate: MoreThanOrEqual(courtDateRange.from) })
-        .andWhere({ courtDate: LessThanOrEqual(courtDateRange.to) })
+        .andWhere({
+          courtDate: MoreThanOrEqual(courtDateRange.from)
+        })
+        .andWhere({
+          courtDate: LessThanOrEqual(courtDateRange.to)
+        })
     }
   }
 
+  if (resolvedByUsername) {
+    let splitResolvedByUsername = resolvedByUsername.replace(/\*|\s+/g, "%")
+
+    if (!splitResolvedByUsername.endsWith("%")) {
+      splitResolvedByUsername = `${splitResolvedByUsername}%`
+    }
```

Comment:
I think this should be refactored into a function as it's used twice, so we can test it independently.

What do you think?